users/script.js

```diff
@@ -474,18 +485,76 @@ function getCheckedValues(groupName) {
 
 function getFilteredUsersURL(checkedValuesSkills, checkedValuesAvailability) {
   const params = new URLSearchParams();
-
   checkedValuesSkills.forEach((skill) => {
     params.append('tagId', skill);
   });
-
   checkedValuesAvailability.forEach((availability) => {
     params.append('state', availability);
   });
-
   return `?${params.toString()}`;
 }
 
+function updateQueryParamsToURL(constructedQueryParam) {
+  const currentURLInstance = new URL(window.location.href);
+  currentURLInstance.search = '';
+  const currentURL = currentURLInstance.href;
+  const newURLWithQueryParams = `${currentURL}${constructedQueryParam}`;
+  window.history.pushState(
+    { path: newURLWithQueryParams },
+    '',
+    newURLWithQueryParams,
+  );
+}
+
+function removeQueryParams() {
+  const currentURLInstance = new URL(window.location.href);
+  currentURLInstance.search = '';
+  const currentURL = currentURLInstance.href;
+  const newUrlWithoutQueryParams = `${currentURL}`;
+  window.history.pushState(
+    { path: newUrlWithoutQueryParams },
+    '',
+    newUrlWithoutQueryParams,
+  );
+}
```

Comment:
- most of the logic inside both the functions look same can we abstract this out into one function and just pass a URL to that ?